cmd/validate.go

```diff
@@ -1,139 +1,178 @@
+			Name:        "token",
+			Description: "token value to use if authentication is required for downloading schema from remote URL",
+		},
+		model.IntFlag{
+			Name:         "max-validation-errors",
+			Description:  "limit the number of errors to output (default 1000, 0 = no limit)",
+			DefaultValue: 1000,
+		},
+		model.IntFlag{
+			Name:         "max-validation-warnings",
+			Description:  "limit the number of warnings to output (default 1000, 0 = no limit)",
+			DefaultValue: 1000,
+		},
+	},
 }
 
-var validateConfigCmd = &cobra.Command{
-	Use:   "config",
-	Short: "Validates a Speakeasy configuration file for SDK generation",
+var validateConfigCmd = &model.Command{
+	Usage: "config",
+	Short: "Validate a Speakeasy configuration file",
 	Long:  `Validates a Speakeasy configuration file for SDK generation.`,
-	RunE:  validateConfig,
+	Run:   validateConfig,
+	Flags: []model.Flag{
+		model.StringFlag{
+			Name:        "dir",
+			Shorthand:   "d",
+			Description: "path to the directory containing the Speakeasy configuration file",
+			Required:    true,
+		},
+	},
 }
 
-func validateInit() {
-	rootCmd.AddCommand(validateCmd)
-	validateOpenAPIInit()
-	validateConfigInit()
+type flags struct {
+	schemaPath string
+	header     string
+	token      string
+	limits     validation.OutputLimits
 }
 
-//nolint:errcheck
-func validateOpenAPIInit() {
-	validateOpenAPICmd.Flags().BoolP("output-hints", "o", false, "output validation hints in addition to warnings/errors")
-	validateOpenAPICmd.Flags().StringP("schema", "s", "", "local filepath or URL for the OpenAPI schema")
-	_ = validateOpenAPICmd.MarkFlagRequired("schema")
-
-	validateOpenAPICmd.Flags().StringP("header", "H", "", "header key to use if authentication is required for downloading schema from remote URL")
-	validateOpenAPICmd.Flags().String("token", "", "token value to use if authentication is required for downloading schema from remote URL")
-
-	validateOpenAPICmd.Flags().Int("max-validation-warnings", 1000, "limit the number of warnings to output (default 1000, 0 = no limit)")
-	validateOpenAPICmd.Flags().Int("max-validation-errors", 1000, "limit the number of errors to output (default 1000, 0 = no limit)")
-
-	validateCmd.AddCommand(validateOpenAPICmd)
-}
-
-func validateConfigInit() {
-	validateConfigCmd.Flags().StringP("dir", "d", "", "path to the directory containing the Speakeasy configuration file")
-	_ = validateConfigCmd.MarkFlagRequired("dir")
-
-	validateCmd.AddCommand(validateConfigCmd)
-}
-
-func validateOpenAPI(cmd *cobra.Command, args []string) error {
-	// no authentication required for validating specs
+func getFlags(f *pflag.FlagSet) (*flags, error) {
+	schemaPath, err := f.GetString("schema")
+	if err != nil {
+		return nil, err
+	}
 
-	schemaPath, err := cmd.Flags().GetString("schema")
+	header, err := f.GetString("header")
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	header, err := cmd.Flags().GetString("header")
+	token, err := f.GetString("token")
```

Comment:
Unfortunately I couldn't come up with a way to improve this interface. Ideally you'd just be able to access all your flags in a well-typed manner without needing to retrieve them from their string names and check the errors

Reply:
I don't actually think this interface is that bad.